src/Transform/ONNX/ConstProp.cpp

```diff
@@ -23,6 +23,8 @@
 #include "llvm/ADT/StringSet.h"
 #include "llvm/Support/Debug.h"
 
+#include "src/Compiler/CompilerOptions.hpp"
+#include "src/Compiler/CompilerPasses.hpp"
```

Comment:
to make it easy to use each of the dialects and passes as libraries that people can link into their own compiler binaries without using the onnx-mlir binary, and to avoid unnecessary cyclical dependencies, we should avoid making the dialects and passes depend on anything in src/Compiler (with #include or cmake LINK_LIBS)

to disable constant propagation, consider instead changing addONNXToMLIRPasses() to not call createConstPropONNXToONNXPass()

Reply:
I originally was going to do it the way you suggested but if I was to do it that way we would need to have multiple `if-statements` in several c++ files (NNPA/CompilerPass/simplifyshaperelatedops/onnxoptransformpass) where as doing it this way requires the update of only the `ConstProp.cpp` but if everyone feels this is much better I can make the update. I am fine with whatever is best for the community :)

Reply:
@sorenlassen Okay I made the updates and tested but I went back to my old way of doing things but it removes the #include "src/Compiler/CompilerOptions.hpp" and #include "src/Compiler/CompilerPasses.hpp" file as you wanted.